### PR TITLE
Implement methods 

### LINKED ISSUE #4: [2pt] Implement methods - Member 1
- Create a Book class in a separate .rb file.
- Create a Label class with an association to the Item class (in a separate .rb file).
- All Book class properties visible in the diagram should be defined and set up in the constructor method.
- All Label class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Label class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Book class
        - should override the method from the parent class
        - should return true if parent's method returns true OR if cover_state equals to "bad"
        - otherwise, it should return false

### PR DESCRIPTION
**Pull request by Wahaj-Ali**
In this PR
-
- Add a Book class in a separate .rb file.
  - Add can_be_archived?() in the Book class 
- Add a Label class with an association to the Item class (in a separate .rb file).
  - Add add_item method in the Label class

### CODE REVIEW

File: Classes/label.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Label
+  attr_accessor :title, :color
+  attr_reader :id, :items
```
Comment (developertariq) [inline Classes/label.rb:3]: With` attr_accessor` we can read and write property value. I think all properties of the Label class should be set up in the constructor method. I suggest you change `attr_accessor :title, :color`  to `attr_reader :title, :color` .

File: Classes/book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require './item'
+
+class Book < Item
+  attr_accessor :publisher, :cover_state
+
```
Comment (developertariq) [inline Classes/book.rb:5]: With attr_accessor we can read and write property value. I think all properties of the Book class should be set up in the constructor method. I suggest you change `attr_accessor :publisher, :cover_state` to `attr_reader :publisher, :cover_state`.

File: Classes/book.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require './item'
+
+class Book < Item
+  attr_accessor :publisher, :cover_state
+
+  def initialize(publisher, cover_state)
+    super(publish_date)
```
Comment (developertariq) [inline Classes/book.rb]: I think `publish_date` should be passed through the constructor as an argument.

**Review by developertariq — CHANGES_REQUESTED** [review]
Hi @Wahaj-Ali  👋
Great job so far, There are some changes that need to be implemented in order to proceed to the next task.

## Highlights
- No linters error.
- Correct github flow.
- Descriptive PR.
## Required Changes
- Please check the comments below.

## [Optional]
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: Classes/label.rb (hunk 1 above)
Comment (Wahaj-Ali) [inline Classes/label.rb:3]: @developertariq Thank you for the suggestions, but all the properties are already defined in the constructor i.e., initialize method in Ruby. attr_accessor is only used to read as well as to write these properties as you already said.

File: Classes/book.rb (hunk 3 above)
Comment (Wahaj-Ali) [reply Classes/book.rb]: I have made this change 😄

File: Classes/book.rb (hunk 2 above)
Comment (Wahaj-Ali) [inline Classes/book.rb:5]: @developertariq thank you for the suggestion, but I have already defined all the properties in the constructor. Can you please recheck? Regards.

**Review by developertariq — APPROVED** [review]
Hi @Wahaj-Ali ,
Your project is complete! There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉

## Highlights
- No linters errors ✔️
- Correct GitHub flow ✔️
- Descriptive PR ✔️
- Good commit history ✔️

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**